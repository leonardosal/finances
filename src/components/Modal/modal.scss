.modal {
  display: none;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.5);

  &.show {
    display: block;
  }
}

.modal-content {
  background-color: #fefefe;
  margin: 15% auto;
  padding: 20px;
  border: 1px solid #888;
  max-width: 480px;

  @media (min-width: 1025px) {
    max-width: 480px;
  }
}

.form {
  display: flex;
  flex-direction: column;

  label {
    padding: 16px 0px;
  }

  input {
    border: 1px solid #ccc;
    height: 30px;
    width: 100%;
    font-size: 1rem;
    display: block;
  }

  select {
    border: 1px solid #ccc;
    height: 30px;
    width: 100%;
    font-size: 1rem;
    display: block;
  }

  .footer {
    display: flex;
    height: 50px;
    justify-content: flex-end;
    align-items: center;
  }

  .field-container {
    padding: 8px 0px;
  }
}
