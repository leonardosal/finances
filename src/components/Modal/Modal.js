import React from 'react';
import propTypes from 'prop-types';
import './modal.scss';

import Button from '../Button';

const Modal = ({
  show,
  toggleModal,
  submitForm,
  onChangeField,
  transaction,
}) => {
  return (
    <div id="modal" className={show ? 'modal show' : 'modal'}>
      <div className="modal-content">
        <h3>Adicionar transação</h3>
        <form
          id="transactionForm"
          name="transactionForm"
          className="form"
          onSubmit={submitForm}
        >
          <label htmlFor="type">
            Tipo
            <select
              name="type"
              id="type"
              onChange={e => onChangeField('type', e.target.value)}
              value={transaction.type}
            >
              <option value="DEBIT">Despesa</option>
              <option value="CREDIT">Receita</option>
            </select>
          </label>
          <label htmlFor="value">
            Valor
            <input
              id="value"
              name="value"
              type="number"
              onChange={e => onChangeField('value', e.target.value)}
              value={transaction.value}
            />
          </label>
          <label htmlFor="description">
            Descrição
            <input
              id="description"
              name="description"
              onChange={e => onChangeField('description', e.target.value)}
              value={transaction.description}
            />
          </label>
          <div className="footer">
            <Button id="btn-save" type="submit" label="Salvar" />
            <Button
              id="btn-cancel"
              type="button"
              label="Cancelar"
              onClick={toggleModal}
            />
          </div>
        </form>
      </div>
    </div>
  );
};

Modal.propTypes = {
  toggleModal: propTypes.func.isRequired,
  submitForm: propTypes.func.isRequired,
  onChangeField: propTypes.func.isRequired,
  show: propTypes.bool,
  transaction: propTypes.shape({
    value: propTypes.number,
    type: propTypes.string,
    description: propTypes.string,
  }),
};

Modal.defaultProps = {
  show: false,
  transaction: {
    value: 0,
    description: '',
    type: 'DEBIT',
  },
};

export default Modal;
