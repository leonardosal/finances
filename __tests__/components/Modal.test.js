import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Modal from '../../src/components/Modal';

it('should render Modal component', () => {
  const props = {
    toggleModal: () => true,
    submitForm: () => true,
    onChangeField: () => true,
    show: false,
    transaction: {
      value: 17.0,
      description: 'Uber',
      type: 'submit',
    },
  };
  const component = mount(
    <Modal
      toggleModal={props.toggleModal}
      submitForm={props.submitForm}
      onChangeField={props.onChangeField}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Modal component when show is true', () => {
  const props = {
    toggleModal: () => true,
    submitForm: () => true,
    onChangeField: () => true,
    show: true,
    transaction: {
      value: 17.0,
      description: 'Uber',
      type: 'submit',
    },
  };
  const component = mount(
    <Modal
      toggleModal={props.toggleModal}
      submitForm={props.submitForm}
      onChangeField={props.onChangeField}
      show={props.show}
      transaction={props.transaction}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Modal component when change values', () => {
  const props = {
    toggleModal: () => true,
    submitForm: () => true,
    onChangeField: () => true,
    show: true,
    transaction: {
      value: 0,
      description: '',
      type: 'DEBIT',
    },
  };
  const component = mount(
    <Modal
      toggleModal={props.toggleModal}
      submitForm={props.submitForm}
      onChangeField={props.onChangeField}
      show={props.show}
      transaction={props.transaction}
    />
  );

  component.find('select#type').simulate('change', {
    target: { value: 'CREDIT' },
  });

  component.find('input#value').simulate('change', {
    target: { value: 4000.0 },
  });

  component.find('input#description').simulate('change', {
    target: { value: 'Salary' },
  });

  expect(toJson(component)).toMatchSnapshot();
});
