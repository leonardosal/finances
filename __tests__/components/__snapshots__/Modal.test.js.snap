// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Modal component 1`] = `
<Modal
  onChangeField={[Function]}
  show={false}
  submitForm={[Function]}
  toggleModal={[Function]}
  transaction={
    Object {
      "description": "",
      "type": "DEBIT",
      "value": 0,
    }
  }
>
  <div
    className="modal"
    id="modal"
  >
    <div
      className="modal-content"
    >
      <h3>
        Adicionar transação
      </h3>
      <form
        className="form"
        id="transactionForm"
        name="transactionForm"
        onSubmit={[Function]}
      >
        <label
          htmlFor="type"
        >
          Tipo
          <select
            id="type"
            name="type"
            onChange={[Function]}
            value="DEBIT"
          >
            <option
              value="DEBIT"
            >
              Despesa
            </option>
            <option
              value="CREDIT"
            >
              Receita
            </option>
          </select>
        </label>
        <label
          htmlFor="value"
        >
          Valor
          <input
            id="value"
            name="value"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
        <label
          htmlFor="description"
        >
          Descrição
          <input
            id="description"
            name="description"
            onChange={[Function]}
            value=""
          />
        </label>
        <div
          className="footer"
        >
          <Button
            id="btn-save"
            label="Salvar"
            onClick={[Function]}
            type="submit"
          >
            <button
              className="button"
              id="btn-save"
              onClick={[Function]}
              type="submit"
            >
              Salvar
            </button>
          </Button>
          <Button
            id="btn-cancel"
            label="Cancelar"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-cancel"
              onClick={[Function]}
              type="button"
            >
              Cancelar
            </button>
          </Button>
        </div>
      </form>
    </div>
  </div>
</Modal>
`;

exports[`should render Modal component when change values 1`] = `
<Modal
  onChangeField={[Function]}
  show={true}
  submitForm={[Function]}
  toggleModal={[Function]}
  transaction={
    Object {
      "description": "",
      "type": "DEBIT",
      "value": 0,
    }
  }
>
  <div
    className="modal show"
    id="modal"
  >
    <div
      className="modal-content"
    >
      <h3>
        Adicionar transação
      </h3>
      <form
        className="form"
        id="transactionForm"
        name="transactionForm"
        onSubmit={[Function]}
      >
        <label
          htmlFor="type"
        >
          Tipo
          <select
            id="type"
            name="type"
            onChange={[Function]}
            value="DEBIT"
          >
            <option
              value="DEBIT"
            >
              Despesa
            </option>
            <option
              value="CREDIT"
            >
              Receita
            </option>
          </select>
        </label>
        <label
          htmlFor="value"
        >
          Valor
          <input
            id="value"
            name="value"
            onChange={[Function]}
            type="number"
            value={0}
          />
        </label>
        <label
          htmlFor="description"
        >
          Descrição
          <input
            id="description"
            name="description"
            onChange={[Function]}
            value=""
          />
        </label>
        <div
          className="footer"
        >
          <Button
            id="btn-save"
            label="Salvar"
            onClick={[Function]}
            type="submit"
          >
            <button
              className="button"
              id="btn-save"
              onClick={[Function]}
              type="submit"
            >
              Salvar
            </button>
          </Button>
          <Button
            id="btn-cancel"
            label="Cancelar"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-cancel"
              onClick={[Function]}
              type="button"
            >
              Cancelar
            </button>
          </Button>
        </div>
      </form>
    </div>
  </div>
</Modal>
`;

exports[`should render Modal component when show is true 1`] = `
<Modal
  onChangeField={[Function]}
  show={true}
  submitForm={[Function]}
  toggleModal={[Function]}
  transaction={
    Object {
      "description": "Uber",
      "type": "submit",
      "value": 17,
    }
  }
>
  <div
    className="modal show"
    id="modal"
  >
    <div
      className="modal-content"
    >
      <h3>
        Adicionar transação
      </h3>
      <form
        className="form"
        id="transactionForm"
        name="transactionForm"
        onSubmit={[Function]}
      >
        <label
          htmlFor="type"
        >
          Tipo
          <select
            id="type"
            name="type"
            onChange={[Function]}
            value="submit"
          >
            <option
              value="DEBIT"
            >
              Despesa
            </option>
            <option
              value="CREDIT"
            >
              Receita
            </option>
          </select>
        </label>
        <label
          htmlFor="value"
        >
          Valor
          <input
            id="value"
            name="value"
            onChange={[Function]}
            type="number"
            value={17}
          />
        </label>
        <label
          htmlFor="description"
        >
          Descrição
          <input
            id="description"
            name="description"
            onChange={[Function]}
            value="Uber"
          />
        </label>
        <div
          className="footer"
        >
          <Button
            id="btn-save"
            label="Salvar"
            onClick={[Function]}
            type="submit"
          >
            <button
              className="button"
              id="btn-save"
              onClick={[Function]}
              type="submit"
            >
              Salvar
            </button>
          </Button>
          <Button
            id="btn-cancel"
            label="Cancelar"
            onClick={[Function]}
            type="button"
          >
            <button
              className="button"
              id="btn-cancel"
              onClick={[Function]}
              type="button"
            >
              Cancelar
            </button>
          </Button>
        </div>
      </form>
    </div>
  </div>
</Modal>
`;
